common/oauthTokenManager.go

```diff
@@ -413,9 +414,14 @@ func (credInfo *OAuthTokenInfo) RefreshTokenWithUserCredential(ctx context.Conte
 		return nil, err
 	}
 
+	var applicationId = credInfo.ClientID
```

Comment:
Would you please move this applicationId default value assigning logic to GetTokenInfo, which is factory method to get OAuthTokenInfo? And in this method, e.g NewServicePrincipalTokenFromManualToken can use ClientId from credInfo, as it now contains ClientId . Thanks!